core/src/test/java/org/apache/druid/data/input/impl/JsonLineReaderTest.java

```diff
@@ -141,7 +141,7 @@ public void testParseRowWithConditional() throws IOException
       while (iterator.hasNext()) {
         final InputRow row = iterator.next();
         Assert.assertEquals("test", Iterables.getOnlyElement(row.getDimension("bar")));
-        Assert.assertEquals(Collections.emptyList(), row.getDimension("foo"));
+        Assert.assertEquals("null", row.getDimension("foo").get(0));
```

Comment:
hmm, why would this change happen? there seems like no 'foo' in the row, a `null` should be turned into an empty list from `getDimension`, not a list with a string `"null"` in it

Reply:
`foo` is in the JSONPathSpec. It comes out as [null] and then `Rows.objectToStrings(event.get(dimension))` makes that. I have added comments in the code